### ISSUE TITLE
Migrate to slackclient 2.x

### ISSUE THREAD

**Issue by lbolla** [pmxbot/pmxbot#85]
`slackclient` 2.x came out and broke backward compatibility.
We need to adapt, by following https://github.com/slackapi/python-slackclient/wiki/Migrating-to-2.x

**Comment by jaraco** [comment]
Unfortunately, those migration docs are pretty naive. They assume a simplistic "hello world" usage. Pmxbot does more than just handle messages. It also has scheduled tasks that need to be run. It also may handle more events that just messages, though I haven't looked.

I started on a draft:

```diff
diff --git a/pmxbot/slack.py b/pmxbot/slack.py
index 61432ce..ef6f4fe 100644
--- a/pmxbot/slack.py
+++ b/pmxbot/slack.py
@@ -17,22 +17,22 @@ log = logging.getLogger(__name__)
 class Bot(pmxbot.core.Bot):
 	def __init__(self, server, port, nickname, channels, password=None):
 		token = pmxbot.config['slack token']
-		sc = importlib.import_module('slackclient')
-		self.slack = sc.SlackClient(token)
+		sc = importlib.import_module('slack')
+		self.slack = sc.RTMClient(token)
 		sr = importlib.import_module('slacker')
 		self.slacker = sr.Slacker(token)
 
 		self.scheduler = schedule.CallbackScheduler(self.handle_scheduled)
 
 	def start(self):
-		res = self.slack.rtm_connect()
-		assert res, "Error connecting"
+		@self.slack.__class__.run_on(event='message')
+		def handle_payload(**payload):
+			self.handle_message(payload)
+
 		self.init_schedule(self.scheduler)
-		while True:
-			for msg in self.slack.rtm_read():
-				self.handle_message(msg)
-			self.scheduler.run_pending()
-			time.sleep(0.1)
+		# todo: build an asyncio handler for scheduled tasks
+
+		self.slack.start()
 
 	def handle_message(self, msg):
 		if msg.get('type') != 'message':
```

So it's a start of an idea, but it's going to require more than just some cursory work, so I suggest we pin to `<2` for now.

**Comment by jaraco** [comment]
Releasing 1122.9.1 now that pins against slackclient 2 until a proper solution can be developed.

**Comment by chrisBrookes93** [comment]
I have a minimal set of changes that gets pmxbot running again after slack turned off their API back in September: https://github.com/pmxbot/pmxbot/pull/99

**Comment by jaraco** [comment]
Fixed in #99 and releasing as v1122.13.
